### PR TITLE
Rename mistral workbook/workflow if name is not equal to action ref

### PR DESCRIPTION
**Pull request by m4dcoder**
If name of the mistral workbook or workflow is not the same as the action ref, rename it before sending the definition to mistral.

### CODE REVIEW

File: st2actions/tests/unit/test_mistral_v2.py [hunk 1]
```diff
@@ -61,6 +62,8 @@
         'workbook_v2_many_workflows_no_default.json',
         'workflow_v2.json',
         'workflow_v2_many_workflows.json',
+        'workbook_v2_rename.json',
+        'workflow_v2_rename.json',
```
Comment (manasdk) [inline st2actions/tests/unit/test_mistral_v2.py]: extra?

File: contrib/examples/actions/mistral-basic-rename.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (manasdk) [inline contrib/examples/actions/mistral-basic-rename.json]: Unrelated to this change but a thought - 

could we infer the action object from the mistral workflow?

File: st2actions/tests/unit/test_mistral_v2.py (hunk 1 above)
Comment (manasdk) [inline st2actions/tests/unit/test_mistral_v2.py]: Ignore. one is workflow other is workbook.

**Comment by manasdk** [comment]
Can you explain why the rename is essential? Would it lead to surprise for the user?

File: contrib/examples/actions/mistral-basic-rename.json (hunk 2 above)
Comment (m4dcoder) [inline contrib/examples/actions/mistral-basic-rename.json]: I'm not sure I understand. Can you rephrase?

**Comment by m4dcoder** [comment]
The mistral API gets name from definition.  So the workflow name needs to be precise.  Otherwise, if user names it other than action ref, the mistral API will save it as whatever the user names it.  On st2 action execution, we assume the workflow is named as action_ref and so we will not be able to execute the workflow.  This is the similar with workbook.  The other option is to throw an error if name differs.
